flexinput/src/main/java/com/lytefast/flexinput/widget/FlexEditText.kt

```diff
@@ -0,0 +1,51 @@
+package com.lytefast.flexinput.widget
+
+import android.content.Context
+import android.support.v13.view.inputmethod.EditorInfoCompat
+import android.support.v13.view.inputmethod.InputConnectionCompat
+import android.support.v13.view.inputmethod.InputContentInfoCompat
+import android.support.v4.os.BuildCompat
+import android.support.v7.appcompat.R
+import android.support.v7.widget.AppCompatEditText
+import android.util.AttributeSet
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputConnection
+
+
+/**
+ * Basic [EditText] component that understands how to consume media from the IME keyboard.
+ *
+ * @author Sam Shih
+ */
+open class FlexEditText @JvmOverloads constructor(
```

Comment:
What effect will this have in clients? Say they want to extend another base class (like if we ever wanna render emojis inline we may need to use a `Fresco` class).

Is there any way we can make this some a helper class an `EditText` can forward the relevant data to in its `onCreateInputConnection`? I am wary of creating base classes in libraries :( If there's no way around this (i.e. if one of those methods is `protected` on `EditText`) oh well.

Reply:
As you can see, the magic is all in the `onCreateInputConnection`. I envision this class more like a simple/default implementation for the handling mechanism for getting media content. If we wanted to use another base class I think it's okay to re-implement the same logic there.

I did consider extracting the logic into a helper function, but:
1) you would still have to override `onCreateInputConnection` in the new class, and call the helper function
2) the logic is not too complex that I'm not sure it's worth extracting

LMK if you disagree with point 2. I can implement as a static function

Reply:
Nope makes sense, thanks!